plugins/scaffolder-backend-module-github-branch-protections/src/actions/githubBranchProtections.ts

```diff
@@ -0,0 +1,237 @@
+            description: `Require branches to be up to date before merging. The default value is 'true'`,
+            type: 'boolean',
+          },
+          requiresCommitSignatures: {
+            title: 'Require commit signing',
+            type: 'boolean',
+            description: `Require commit signing so that you must sign commits on this branch.`,
+          },
+          requiresConversationResolution: {
+            title: 'Required Conversation Resolution',
+            description:
+              'Requires all conversations on code to be resolved before a pull request can be merged into this branch',
+            type: 'boolean',
+          },
+          branches: {
+            title: 'Required list of branch patterns.',
+            description:
+              'List of branch protection patterns to apply the protections onto.',
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+          isAdminEnforced: {
+            title: 'Enforce Admins On Protected Branches',
+            type: 'boolean',
+            description: `Enforce admins to adhere to default branch protection. The default value is 'true'`,
+          },
+          token: {
+            title: 'Authentication Token',
+            type: 'string',
+            description: 'The token to use for authorization to GitHub',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const {
+        repoUrl,
+        requiresCodeOwnerReviews = true,
+        dismissStaleReviews = true,
+        requiresApprovingReviews = true,
+        requiredApprovingReviewCount = 1,
+        requiresStatusChecks = false,
+        requiredStatusCheckContexts = [],
+        requiresBranchesToBeUpToDate = false,
+        requiresConversationResolution = false,
+        requiresCommitSignatures = false,
+        branches = [],
+        isAdminEnforced = true,
+        token: providedToken,
+      } = ctx.input;
+
+      const { owner, repo } = parseRepoUrl(repoUrl, integrations);
+
+      if (!owner) {
+        throw new InputError('Invalid repository owner provided in repoUrl');
+      }
+
+      if (!branches || branches.length <= 0) {
+        throw new InputError('The branches field is empty');
+      }
+
+      const client = new Octokit(
+        await getOctokitOptions({
+          integrations,
+          credentialsProvider: githubCredentialsProvider,
+          repoUrl: repoUrl,
+          token: providedToken,
+        }),
+      );
+
+      const sanitizedRepo = repo.replace(/ /g, '-');
+
+      const repository: RepoQuery = await client.graphql(`
+        {
+          repository(followRenames: true, owner:"${owner}", name:"${sanitizedRepo}") {
+            id
+          }
+        }
+      `);
+
+      if (!repository?.repository?.id) {
+        throw new NotFoundError('Input repository does not exist');
+      }
+
+      branches.forEach(async (branchName: string) => {
```

Comment:
if you do this, `handler` will always succeed without waiting until all the `client.graphql` calls have finished their work.
I guess you need to use a for each if your intention is to invoke `client.graphql` sequentially

Reply:
ah good point! Thanks @vinzscam, I will refactor

Reply:
mmm I don't think the issue is resolved as you are still using a `.forEach`